### ISSUE TITLE
OS crash after getting bubble notification

### ISSUE THREAD

**Issue by blaugrau3001** [signalapp/Signal-Android#13285]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
My OS crashed after getting a notification and opening it in the android conversation bubble feature

### Steps to reproduce
- get a message with notification
- try to open the message in a notification bubble
- than the crash happend

**Actual result:** The OS began to glitch at the lockscreen and going black and after 10s the phone shut down. After booting I got an error from the bootloader stating that the OS is corrupted, but after another reboot everything was fine.
**Expected result:** The message bubble should show up with the chat

### Device info

**Device:** Pixel 7
**Android version:** 14 Grapheneos UP1A.231105.003.2023111500
**Signal version:** Version 6.41.0-nightly-2023-11-13-16-00-20

### Link to debug log
 https://debuglogs.org/android/6.41.0-nightly-2023-11-13-16-00-20/7a6379d279b8470d772fbebe7a46c30dd58f45239b1686786a6a98ecb262f353

**Comment by maximko** [comment]
Same here. I have old oneplus phone with old android but latest signal from google play. I found that crash only triggers if message contains an image (can be web preview too) and "Screen lock" option is set. com.android.systemui goes crazy and shuts the phone down.

**Comment by blaugrau3001** [comment]
> Same here. I have old oneplus phone with old android but latest signal from google play. I found that crash only triggers if message contains an image (can be web preview too) and "Screen lock" option is set. Just receiving a message is enough to crash, no interaction needed. Crash happens even if phone is locked. com.android.systemui goes crazy and shuts the phone down. I have system-wide logcat captured in the time of crash and I can share it privately.

I think you issue is more related to this https://github.com/signalapp/Signal-Android/issues/11828 , because your phone didn't completely crashed and you probably don't use the nightly build, right?

**Comment by maximko** [comment]
Indeed I don't use nightly but the outcome is the same — ui began to glitch and phone reboots to bootloader.

**Comment by hondogitsune** [comment]
I don't think this related to notifications at all, please read my analysis https://github.com/signalapp/Signal-Android/issues/13287 .

> only triggers if message contains an image 

This is a correct assertion. Please provide the images as samples in my issue https://github.com/signalapp/Signal-Android/issues/13287 , or at least their ICC profile by viewing them in a hex editor, if the image is too private.

**Comment by nitin065** [comment]
I'm facing the same issue. 

I tried to get the log dump from my phone and could see NPE reading Image notification. Here is the log excerpt:

```
========================================
2023-11-26 22:04:44 system_app_crash (text, 1656 bytes)
Process: com.android.systemui
PID: 12764
UID: 10005
Flags: 0x30c83e0d
Package: com.android.systemui v29 (10)
Foreground: No
Build: <<removed>>

java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:399)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void java.io.InputStream.close()' on a null object reference
	at com.android.internal.widget.LocalImageResolver.getBoundsOptionsForImage(LocalImageResolver.java:67)
	at com.android.internal.widget.LocalImageResolver.resolveImage(LocalImageResolver.java:39)
	at com.android.systemui.statusbar.notification.row.NotificationInlineImageResolver.resolveImage(NotificationInlineImageResolver.java:76)
	at com.android.systemui.statusbar.notification.row.NotificationInlineImageCache$PreloadImageTask.doInBackground(NotificationInlineImageCache.java:99)
	at com.android.systemui.statusbar.notification.row.NotificationInlineImageCache$PreloadImageTask.doInBackground(NotificationInlineImageCache.java:86)
	at android.os.AsyncTask$3.call(AsyncTask.java:378)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
```

**Comment by spacesynth** [comment]
> I have now a more detailed log
> 
> ```
> Caused by: java.lang.SecurityException: Permission Denial: opening provider org.thoughtcrime.securesms.providers.AvatarProvider from ProcessRecord{e78c3 29944:com.android.systemui/u0a106} (pid=29944, uid=10106) that is not exported from UID 10193
> ```
> 
> The error is most likely caused by grapheneos, but I am not an expert. But what I can say is that the error is caused not by a image instead by the bubble notification.
> 

And an avatar is what?

**Comment by blaugrau3001** [comment]
> > I have now a more detailed log
> > 
> > ``` 
> > Permission Denial: opening provider org.thoughtcrime.securesms.providers.AvatarProvider
> > ```
> > 
> > The error is most likely caused by grapheneos, but I am not an expert. But what I can say is that the error is caused not by a image instead by the bubble notification.
> >
> 
> And an avatar is what?
> 

The contact doesnt have a avatar

**Comment by zenitraM** [comment]
I've started suffering this bug, not with Signal but with a specific conversation in Telegram that I'd accidentally enabled chat bubbles for, on stock Android 14 on my Pixel 7. 
So it seems to be an Android 14 system related issue entirely, or somehow it is something that Telegram is also able to trigger.

**Comment by blaugrau3001** [comment]
> I've started suffering this bug, not with Signal but with a specific conversation in Telegram that I'd accidentally enabled chat bubbles for, on stock Android 14 on my Pixel 7. So it seems to be an Android 14 system related issue entirely, or somehow it is something that Telegram is also able to trigger.

You are right on the grapheneos issue someone wrote that this is a AOSP bug

**Comment by DinoGoEatChu** [comment]
Any update or fix for this? I am having the same issue

**Comment by blaugrau3001** [comment]
> Any update or fix for this? I am having the same issue 

Its working fine now, I am on the Android 14 QPR1 build, it seems that it got fixed

**Comment by gooomis** [comment]
> > Any update or fix for this? I am having the same issue
> 
> Its working fine now, I am on the Android 14 QPR1 build, it seems that it got fixed Signal Version: 6.43.0-nightly-2023-12-15-17-00-17

Great news! Quick question tho - how'd you get the nightly version?

**Comment by blaugrau3001** [comment]
> > > Any update or fix for this? I am having the same issue
> > 
> > 
> > Its working fine now, I am on the Android 14 QPR1 build, it seems that it got fixed Signal Version: 6.43.0-nightly-2023-12-15-17-00-17
> 
> Great news! Quick question tho - how'd you get the nightly version?

No Problem
https://community.signalusers.org/t/public-username-testing-staging-environment/56866
But otherwise just use the beta feature of the playstore if you didnt meant the username thing
